src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1382,13 +1382,15 @@ private bool IsQueryMatch(string query, string symbolName)
             // TODO Should be using dynamic registrations
             if (!this.currentSettings.CodeFolding.Enable) { return null; }
             var result = new List<FoldingRange>();
+            ScriptFile script = editorSession.Workspace.GetFile(documentUri);
+            // If we're showing the last line, decrement the Endline of all regions by one.
+            int endLineOffset = this.currentSettings.CodeFolding.ShowLastLine ? -1 : 0;
             foreach (FoldingReference fold in TokenOperations.FoldableRegions(
-                editorSession.Workspace.GetFile(documentUri).ScriptTokens,
-                this.currentSettings.CodeFolding.ShowLastLine))
+                script.ScriptTokens).ToArray())
```

Comment:
Could you assign `TokenOperations.FoldableRegions(script.ScriptTokens).ToArray()` to an intermediate variable (`foldableRegions`?) for better debuggability and so this foreach doesn't have to wrap to a second line?

Reply:
BTW is it really necessary to create/copy a new array here - which I see the ToArray method below is doing?  The Dictionary.Values property will return an `IEnumerable<FoldingReference>` which is all you need to iterate (foreach) over it.

Reply:
There are serialization errors when passing `IEnumberable<FoldingReference>` because the LSP needs `FoldingRange[]` (https://github.com/PowerShell/PowerShellEditorServices/blob/fbe88715a9b20ecc31a6827756f58225a336ded1/src/PowerShellEditorServices.Protocol/LanguageServer/Folding.cs#L40)

And by errors I mean the serialization does not conform to the LSP specification.  Which is why I do an object conversion.  I tried to keep the number of array creations/copies to a minimum

Reply:
> Could you assign TokenOperations.FoldableRegions(script.ScriptTokens).ToArray() to an intermediate variable (foldableRegions?) for better debuggability and so this foreach doesn't have to wrap to a second line?

I can.  I was trying to avoid creating array copies.

Reply:
Using intermediates variables won't cause array copies.

> There are serialization errors when passing IEnumberable<FoldingReference>

Reply:
Fixed.  Went with
`foreach (KeyValuePair<int, FoldingReference> entry in TokenOperations.FoldableRegions(script.ScriptTokens))`

Reply:
When moving from inheritance to composition this has now changed to;
`foreach (FoldingReference fold in TokenOperations.FoldableReferences(script.ScriptTokens).References)`